### PR TITLE
fix: fixes bug where files under 512 bytes would cause EOF error

### PR DESCRIPTION
**Pull request by colek42**
This fixes the EOF error and some additional issues relating to small files discovered when I created some tests.

I added support for detecting pdf files to show the pattern which can be used when adding additional type detection.

### CODE REVIEW

File: attestation/product/product_test.go [hunk 1]
```diff
@@ -15,7 +15,12 @@
 package product
 
 import (
+	"archive/tar"
 	"crypto"
+	"io/ioutil"
```
Comment (mikhailswift) [inline attestation/product/product_test.go:20]: io/ioutil is deprecated

File: attestation/product/product.go [hunk 2]
```diff
@@ -126,15 +132,69 @@ func (a *Attestor) Subjects() map[string]cryptoutil.DigestSet {
 }
 
 func getFileContentType(file *os.File) (string, error) {
-	// Only the first 512 bytes are used to sniff the content type.
-	buffer := make([]byte, 512)
-	_, err := file.Read(buffer)
```
Comment (mikhailswift) [inline attestation/product/product.go:131]: This code was fine, why not just check for EOF and not return in that case?

File: attestation/product/product.go [hunk 3]
```diff
@@ -126,15 +132,69 @@ func (a *Attestor) Subjects() map[string]cryptoutil.DigestSet {
...
 	}
+	fileSize := fileInfo.Size()
 
-	// Use the net/http package's handy DectectContentType function. Always returns a valid
-	// content-type by returning "application/octet-stream" if no others seemed to match.
+	// Read up to 512 bytes from the file.
+	bufferSize := int(math.Min(float64(fileSize), 512))
+	buffer := make([]byte, bufferSize)
+	_, err = file.Read(buffer)
+	if err != nil && err != io.EOF {
```
Comment (mikhailswift) [inline attestation/product/product.go:146]: As above, you could just use file.Read and check for EOF as you are here.

**Review by mikhailswift — CHANGES_REQUESTED** [review]

**Comment by mikhailswift** [comment]
There's a bit more changes here than just fixing the EOF error not being handled when the file is less than 512 bytes, should this be separated into a separate PR or can you explain a bit about what spawned the additional changes?

**Comment by colek42** [comment]
When I added the tests, I found some edge cases where file content detection was not done correctly with the HTTP package. especially when the file was under 512 bytes

**Review by mikhailswift — APPROVED** [review]
